plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/GoogleLoginService.java

```diff
@@ -54,21 +72,16 @@
    * operation and display a general message that login is required but was cancelled or failed.
    *
    * Must be called from a UI context.
-   *
-   * @param shellProvider provides a shell for the login screen if login is necessary
-   * @throws IOException can be thrown by the underlying Login API request (e.g., network
-   *     error from the transport layer while sending/receiving a HTTP request/response.)
    */
-  public Credential getActiveCredential(IShellProvider shellProvider) throws IOException {
-    Credential credential = getCachedActiveCredential();
-
-    if (credential == null) {
-      credential = logIn(shellProvider);
-
-      IEclipseContext eclipseContext = PlatformUI.getWorkbench().getService(IEclipseContext.class);
-      eclipseContext.set(STASH_OAUTH_CRED_KEY, credential);
+  public Credential getActiveCredential() {
+    // TODO: holding a lock for a long period of time (especially when waiting for UI events)
```

Comment:
@briandealwis Yeah, that'd be cool. The problem here is that the entire stream of login work is done by the ide-login library we can't control and the library mixes UI flow and non-UI flow in itself.